codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FrameCodec.java

```diff
@@ -420,26 +420,31 @@ private void writeHeadersFrame(
             // We should not re-use ids.
             assert old == null;
 
-            // Clean up the stream being initialized if writing the headers fails.
-            promise.addListener(new ChannelFutureListener() {
-                @Override
-                public void operationComplete(ChannelFuture channelFuture) {
-                    if (!channelFuture.isSuccess()) {
-                        frameStreamToInitializeMap.remove(streamId);
-                    }
-                }
-            });
-
             encoder().writeHeaders(ctx, streamId, headersFrame.headers(), headersFrame.padding(),
                     headersFrame.isEndStream(), promise);
 
             if (!promise.isDone()) {
                 numBufferedStreams++;
                 promise.addListener(bufferedStreamsListener);
```

Comment:
Can also get rid of `bufferedStreamsListener` now, and just add `numBufferedStreams--` to the new listener

Reply:
Done